### PR TITLE
Add evidence, don't collapse inline

### PR DESCRIPTION
**Pull request by xwatkins**
## Purpose
Address issues due to the rendering of keywords and their evidences

## Approach
- Added the evidence tag component
- Changed logic to not use ExpandableList when in inline mode
- Remove extraneous dot

## Testing
Update test snapshots

## Checklist
- [ ] My PR is scoped properly, and “does one thing only”
- [x] I have reviewed my own code
- [x] I have checked that linting checks pass and type safety is respected
- [x] I have checked that tests pass and coverage has at least improved, and if not explained the reasons why

### CODE REVIEW

File: src/uniprotkb/components/results/UniProtKBCard.tsx [hunk 1]
```diff
@@ -34,11 +34,12 @@ const UniProtKBCard = ({ data }: { data: UniProtkbAPIModel }) => {
       'Disease',
     ]);
 
-    return categorisedKeywords.map((keywordCategory, index) => (
-      <Fragment key={keywordCategory.category}>
-        {index > 0 && ' · '}
-        <KeywordList keywords={keywordCategory.keywords} inline />
-      </Fragment>
+    return categorisedKeywords.map((keywordCategory) => (
```
Comment (dlrice) [inline src/uniprotkb/components/results/UniProtKBCard.tsx:37]: Screenshot 2021-12-01 at 14 28 42

Noticed some slight spacing differences between each of the `KeywordList` components - maybe would be easiest to flatten all of the `categorisedKeywords` and use a single `KeywordList`

Comment (xwatkins) [reply src/uniprotkb/components/results/UniProtKBCard.tsx:37]: Good catch - in the end removed the `useMemo` altogether as we're not giving any indication of category, just throwing keywords on the screen.

Comment (dlrice) [reply src/uniprotkb/components/results/UniProtKBCard.tsx]: Without using `getKeywordsForCategories` we aren't filtering to only include:
```
      'Molecular function',
      'Biological process',
      'Disease',
```
but will have all:
```
export type KeywordCategory =
  | 'Biological process'
  | 'Cellular component'
  | 'Coding sequence diversity'
  | 'Developmental stage'
  | 'Disease'
  | 'Domain'
  | 'Ligand'
  | 'Molecular function'
  | 'PTM'
  | 'Technical term';
```
Seems we intentionally filtered this in the past - do we still want to do this?

**Comment by dlrice** [comment]
Screenshot 2021-12-01 at 15 06 24

File: src/uniprotkb/components/results/UniProtKBCard.tsx (hunk 1 above)
Comment (xwatkins) [reply src/uniprotkb/components/results/UniProtKBCard.tsx]: Right, good point I thought it was like aspects - I'll revert and flatten.

Comment (aurel-l) [reply src/uniprotkb/components/results/UniProtKBCard.tsx]: You could also just filter yourself, something like "keywords.filter(kw => selectedCategories.has(kw.category))" with selectedCategories the set of 3 categories, defined outside of the component, and you should be good to go.

**Comment by aurel-l** [comment]
> Screenshot 2021-12-01 at 15 06 24

See `.keyword-view` class, space needs to be after except on the last element to avoid .1 raised by @dlrice. I think .2 is OK, cards could have variable heights depending on content, and it would avoid having more things scrolling horizontally

**Comment by dlrice** [comment]
> I think .2 is OK, cards could have variable heights depending on content, and it would avoid having more things scrolling horizontally

Was thinking more about keeping a single keyword together on one line eg:

one keyword on one line:
```
#keyword1 #keyword2
#keyword3 #Protein transport
```
one keyword split over two lines:
```
#keyword1 #keyword2
#keyword3  #Protein
transport
```

**Comment by xwatkins** [comment]
Switched to flexbox as it tackles both issues.

File: src/uniprotkb/components/results/UniProtKBCard.tsx [hunk 2]
```diff
@@ -23,23 +23,18 @@ const UniProtKBCard = ({ data }: { data: UniProtkbAPIModel }) => {
...
-
-    return categorisedKeywords.map((keywordCategory, index) => (
-      <Fragment key={keywordCategory.category}>
-        {index > 0 && ' · '}
-        <KeywordList keywords={keywordCategory.keywords} inline />
-      </Fragment>
-    ));
+    ])
+      .map(({ keywords }) => keywords)
+      .flat();
```
Comment (dlrice) [inline src/uniprotkb/components/results/UniProtKBCard.tsx:37]:
```suggestion
      .flatMap(({ keywords }) => keywords);
```

**Review by dlrice — APPROVED** [review]

**Pull request merged**